src/sql/src/parsers/tql_parser.rs

```diff
@@ -66,121 +96,184 @@ impl<'a> ParserContext<'a> {
+                        )));
+                    }
+                };
+                Ok(value)
             }
+            _ => Self::parse_tokens(tokens),
         };
-        parser.expect_token(&token)?;
+        parser.expect_token(&delimiter_token)?;
+        result
+    }
 
-        Ok(value)
+    fn parse_tokens(tokens: Vec<Token>) -> std::result::Result<String, TQLError> {
+        Self::parse_to_expr(tokens)
+            .and_then(Self::parse_to_logical_expr)
+            .and_then(Self::simplify_expr)
+            .and_then(Self::evaluate_expr)
     }
 
-    fn parse_tql_query(
-        parser: &mut Parser,
-        sql: &str,
-        delimiter: &str,
-    ) -> std::result::Result<String, ParserError> {
-        let index = sql.to_uppercase().find(delimiter);
-
-        if let Some(index) = index {
-            let index = index + delimiter.len() + 1;
-            if index >= sql.len() {
-                return Err(ParserError::ParserError("empty TQL query".to_string()));
-            }
+    fn parse_to_expr(tokens: Vec<Token>) -> std::result::Result<sqlparser::ast::Expr, TQLError> {
+        Parser::new(&GreptimeDbDialect {})
+            .with_tokens(tokens)
+            .parse_expr()
+            .map_err(|err| TQLError::Parser(format!("Failed to convert to expression: {err:?}")))
+    }
 
-            let query = &sql[index..];
+    fn parse_to_logical_expr(expr: sqlparser::ast::Expr) -> std::result::Result<Expr, TQLError> {
+        let empty_df_schema = DFSchema::empty();
+        SqlToRel::new(&StubContextProvider {})
+            .sql_to_expr(expr.into(), &empty_df_schema, &mut Default::default())
+            .map_err(|err| {
+                TQLError::Parser(format!("Failed to convert to logical expression {err:?}"))
+            })
+    }
 
-            while parser.next_token() != Token::EOF {
-                // consume all tokens
-                // TODO(dennis): supports multi TQL statements separated by ';'?
-            }
+    fn simplify_expr(logical_expr: Expr) -> std::result::Result<Expr, TQLError> {
+        let empty_df_schema = DFSchema::empty();
+        let execution_props = ExecutionProps::new().with_query_execution_start_time(Utc::now());
+        let info = SimplifyContext::new(&execution_props).with_schema(Arc::new(empty_df_schema));
+        ExprSimplifier::new(info)
+            .simplify(logical_expr)
+            .map_err(|err| {
+                TQLError::Simplification(format!("Failed to simplify expression {err:?}"))
+            })
+    }
 
-            // remove the last ';' or tailing space if exists
-            Ok(query.trim().trim_end_matches(';').to_string())
-        } else {
-            Err(ParserError::ParserError(format!("{delimiter} not found",)))
+    fn evaluate_expr(simplified_expr: Expr) -> std::result::Result<String, TQLError> {
+        match simplified_expr {
+            Expr::Literal(ScalarValue::TimestampNanosecond(v, _))
```

Comment:
Maybe we can support other timestamp types too, such as `TimestampMillisecond` and `TimestampSecond` etc.